core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -2915,6 +2966,84 @@ wasi_ssp_sock_bind(
     return __WASI_ESUCCESS;
 }
 
+__wasi_errno_t
+wasi_ssp_addr_resolve(
+#if !defined(WASMTIME_SSP_STATIC_CURFDS)
+    struct fd_table *curfds,
+#endif
+    const char *host, const char *service, __wasi_addr_info_hints_t *hints,
+    __wasi_addr_info_t *addr_info, __wasi_size_t addr_info_size,
+    __wasi_size_t *max_size)
+{
+    struct addrinfo posix_hints, *res, *result;
+    __wasi_size_t pos = 0;
+    int ret;
+
+    if (hints->hints_enabled) {
+        memset(&posix_hints, 0, sizeof(posix_hints));
+        posix_hints.ai_family = hints->family == INET4 ? PF_INET : PF_INET6;
+        posix_hints.ai_socktype =
+            hints->type == SOCKET_STREAM ? SOCK_STREAM : SOCK_DGRAM;
+    }
+
+    ret = getaddrinfo(host, strlen(service) == 0 ? NULL : service,
+                      hints->hints_enabled ? &posix_hints : NULL, &result);
+    if (ret != 0) {
+        return convert_getaddrinfo_error_to_wasi_error_code(ret);
+    }
+
+    res = result;
+    while (res) {
```

Comment:
maybe something like:
```suggestion
    while (res && addr_info_size > pos)) {
```

Reply:
We want to go through the full list to know its size (so we can return it through `max_info_size` parameter).

Reply:
OK.